### PR TITLE
CONSOLE-2384: Add metrics tab to select resources

### PR DESCRIPTION
**Pull request by TheRealJon**
Add new metrics tab to Deployment, StatefulSet, and DaemonSet details pages and move Build, Pod, and Route metrics to a separate tab for consistency.

### Deployment metrics
deployment-metrics

### Build metrics
build metrics

### Pod metrics
pod metrics

### Route metrics
route metrics

### CODE REVIEW

**Comment by spadgett** [comment]
cc @openshift/openshift-team-monitoring

File: frontend/packages/console-shared/src/components/dashboard/dashboard-card/card.scss
Comment (spadgett) [inline frontend/packages/console-shared/src/components/dashboard/dashboard-card/card.scss:11]: Should this be `co-dashboard-card__resource-metrics`? It feels like an element rather than a modifier.

http://getbem.com/naming/

It would be better if we can add a specific class name to the wrapper element here. (I'm not sure if we are able to pass a `className` through to the nested component).

File: frontend/packages/console-shared/src/promql/resource-metrics.ts
Comment (spadgett) [inline frontend/packages/console-shared/src/promql/resource-metrics.ts:148]: nit: We could determine the type from `obj.apiVersion` and `obj.kind` if we wanted to remove one of the function arguments.

**Review by spadgett — COMMENTED** [review]
Nice work! No major comments. We can follow up with a separate story for other workload types.

/assign @yapei @ahardin-rh @sferich888 
for approvals

**Comment by spadgett** [comment]
We should probably move pod, build, and route metrics under a tab as part of this change so we're consistent.

**Comment by TheRealJon** [comment]
@spadgett I've updated the PR to move Build, Pod, and Route metrics into new tabs as well

**Comment by yapei** [comment]
@yanpzhan is testing the PR and will report test results back

**Comment by yanpzhan** [comment]
@TheRealJon  Tested against the pr, and drafted a case which was pasted in https://issues.redhat.com/browse/CONSOLE-2384.
For now the metrics tab page work well, will approve the pr if there is no other checkpoint need to be tested.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals

**Comment by spadgett** [comment]
> @TheRealJon Tested against the pr, and drafted a case which was pasted in https://issues.redhat.com/browse/CONSOLE-2384.
> For now the metrics tab page work well, will approve the pr if there is no other checkpoint need to be tested.

Thanks, @yanpzhan. I've approved the test cases. Based on this comment, I'm adding the qe-approved label.

/label qe-approved

**Pull request merged**
